api/cl_khr_command_buffer.asciidoc

```diff
@@ -110,6 +100,28 @@ following reasons:
   other extensions layered on top to take advantage of them to provide additional
   mutable functionality.
 
+==== Command Synchronization
+
+Device-side {cl_sync_point_khr_TYPE} synchronization-points can be used within
+command-buffers to define command dependencies. This allows the commands of a
+command-buffer to execute out-of-order on a single <<compatible, compatible>>
+command-queue. The command-buffer itself has no inherent in-order/out-of-order
+property, this ordering is inferred from the command-queue used on command
+recording. Out-of-order enqueues without event dependencies of both regular
+commands, such as {clEnqueueFillBuffer}, and command-buffers are allowed to
+execute concurrently, and it is up to the user to express any dependencies using
+events.
```

Comment:
I realize this text isn't new and just moved, but when I read this sentence as part of this review I initially thought it was saying that an event dependency is required in all cases to express dependencies.  After re-reading, I think it's only intending to apply to out-of-order queues though, correct?  Not sure if it'd help to have more words or fewer, or if we're OK with the text as-is, mostly noting for completeness.

Reply:
If it's not clear seems like a good opportunity to try improve the text. How about?

> {clEnqueueCommandBuffer} submissions to an out-of-order queue have the same execution semantics are other operations enqueued to an out-of-order queue, such as {clEnqueueFillBuffer}. Where execution between enqueued operations may happen concurrently unless dependencies between the operations are expressed with events.

Reply:
I've pushed that wording as a separate commit, so it can be reviewed standalone.